src/openfermion/measurements/_bayesian_estimators.py

```diff
@@ -0,0 +1,845 @@
+                cos(n*phi_j + beta).
+            vector_guess (numpy array): a prior estimate of the function,
+                given in terms of the Fourier components c_{j,n}.
+                If None, estimated as a flat function over [-pi,pi].
+
+        Raises:
+            ValueError: if prior distribution does not have required shape.
+            ValueError: if insufficient amplitude guesses are given.
+            ValueError: if starting amplitudes do not break symmetry.
+        """
+
+        # Store size data
+        self._num_vectors = num_vectors
+        self._num_freqs = num_freqs
+
+        if amplitude_guess is None:
+            amplitude_guess = [1]
+        if amplitude_vars is None:
+            amplitude_vars = [[0]]
+
+        # Store prior information
+        if vector_guess is not None and (
+                num_vectors != vector_guess.shape[1]
+                or 2*num_freqs+1 != vector_guess.shape[0]):
+            raise ValueError('''Prior distribution of phases has shape {},
+                required shape is {}'''.format(
+                    vector_guess.shape, (2*num_freqs+1, num_vectors)))
+
+        self._vector_guess = vector_guess
+
+        if num_vectors != len(amplitude_guess):
+            raise ValueError('''I need {} amplitudes, but you have
+                given me {}.'''.format(num_vectors, len(amplitude_guess)))
+
+        self._amplitude_guess = amplitude_guess
+        self._amplitude_vars = amplitude_vars
+        self._amplitude_estimates = None
+        self._fourier_vectors = None
+
+        if any([numpy.isclose(self._amplitude_guess[j],
+                              self._amplitude_guess[j+1])
+                for j in range(num_vectors - 1)]):
+            raise ValueError('''Starting amplitudes must be different.''')
+
+        # Store max_n
+        self._max_n = max_n
+
+        # Create matrices for multiplication
+        self._make_matrices()
+        self.reset()
+
+    def reset(self):
+        """
+        Resets the distribution.
+        """
+        if self._vector_guess is not None:
+            self._fourier_vectors = numpy.array(self._vector_guess)
+        else:
+            # Initialize probability distributions
+            self._fourier_vectors = numpy.zeros((self._num_freqs*2+1,
+                                                 self._num_vectors))
+            for j in range(self._num_vectors):
+                self._fourier_vectors[0, j] = 1
+
+        self._amplitude_estimates = numpy.array(self._amplitude_guess)
+
+    def get_real_dist(self, x_vec=None):
+        """
+        Generates the real distribution over a sequence of points
+
+        Args:
+            x_vec (numpy array or list): the points at which the distribution is generated.
+                If None, defaults to 101 points between -pi and pi.
+        Returns:
+            y_vecs (list): the value of the distributions at the given x points.
```

Comment:
Return a numpy array instead and explicitly document that it is two dimensional, and what the rows and columns are.